internal/pkg/storageos/client_test.go

```diff
@@ -0,0 +1,199 @@
+package storageos
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/golang/mock/gomock"
+	api "github.com/storageos/go-api/v2"
+
+	"github.com/storageos/api-manager/internal/pkg/storageos/mocks"
+)
+
+func TestClient_Refresh(t *testing.T) {
+	secretPath, err := ioutil.TempDir("", "secrets")
+	if err != nil {
+		t.Fatalf("failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(secretPath)
+	if err := ioutil.WriteFile(filepath.Join(secretPath, "username"), []byte("username"), 0644); err != nil {
+		t.Fatalf("failed to write username file: %v", err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(secretPath, "password"), []byte("password"), 0644); err != nil {
+		t.Fatalf("failed to write password file: %v", err)
+	}
```

Comment:
Since this is a time based test, it'd be better to run them in parallel. How about moving the above in the `t.Run()`, to avoid tests from conflicting with each other, and enable `t.Parallel()`?